integrations/opentelemetry-wsgi/tests/test_wsgi_middleware.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import io
+import unittest
+import unittest.mock as mock
+from opentelemetry_wsgi import OpenTelemetryMiddleware
+from opentelemetry import trace as trace_api
+
+
+class Response:
+    def __init__(self):
+        self.iter = iter([b"*"])
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return next(self.iter)
+
+
+def simple_wsgi(environ, start_response):
+    start_response("200 OK", [("Content-Type", "text/plain")])
+    return [b"*"]
+
+
+def iter_wsgi(environ, start_response):
+    start_response("200 OK", [("Content-Type", "text/plain")])
+    return Response()
+
+
+def error_wsgi(environ, start_response):
+    try:
+        raise ValueError
+    except ValueError:
+        exc_info = sys.exc_info()
+    start_response("200 OK", [("Content-Type", "text/plain")], exc_info)
+    exc_info = None
+    return [b"*"]
+
+
+class TestWsgiMiddleware(unittest.TestCase):
```

Comment:
I think this a great argument why we should use pytest.

This is a great class that provides a lot of functionality, but we can't consume it in child tests that might benefit from a constructed wsgi app. But we can't use it elsewhere without subclassing, and multiple inheritance doesn't work for unittest.TestCase (because there's only one setUp that's called at the end of the day).

Just a side note, not something that needs to be addressed in this PR.

Reply:
I ❤️ pytest! 😄